pennylane/devices/default_qubit_pt.py

```diff
@@ -0,0 +1,214 @@
+            QNode differentiation method is not supported and it is recommended to consider
+            switching device to ``default.qubit`` and using ``diff_method="parameter-shift"``.
+    """
+
+    name = "Default qubit (PyTorch) PennyLane plugin"
+    short_name = "default.qubit.torch"
+
+    parametric_ops = {
+        "PhaseShift": pt_ops.PhaseShift,
+        "ControlledPhaseShift": pt_ops.ControlledPhaseShift,
+        "RX": pt_ops.RX,
+        "RY": pt_ops.RY,
+        "RZ": pt_ops.RZ,
+        "Rot": pt_ops.Rot,
+        "MultiRZ": pt_ops.MultiRZ,
+        "CRX": pt_ops.CRX,
+        "CRY": pt_ops.CRY,
+        "CRZ": pt_ops.CRZ,
+        "CRot": pt_ops.CRot,
+        "SingleExcitation": pt_ops.SingleExcitation,
+        "SingleExcitationPlus": pt_ops.SingleExcitationPlus,
+        "SingleExcitationMinus": pt_ops.SingleExcitationMinus,
+        "DoubleExcitation": pt_ops.DoubleExcitation,
+        "DoubleExcitationPlus": pt_ops.DoubleExcitationPlus,
+        "DoubleExcitationMinus": pt_ops.DoubleExcitationMinus,
+    }
+
+    C_DTYPE = torch.complex128
+    R_DTYPE = torch.float64
+    _asarray = staticmethod(torch.as_tensor)
+    _dot = staticmethod(lambda x, y: torch.tensordot(x, y, axis=1))
+    _abs = staticmethod(torch.abs)
+    _reduce_sum = staticmethod(torch.sum)
+    _reshape = staticmethod(torch.reshape)
+    ## TODO:  Need to verify reshape
+    _flatten = staticmethod(lambda tensor: torch.reshape(tensor, [-1]))
+    _gather = staticmethod(torch.gather)
+    _einsum = staticmethod(torch.einsum)
+    #_cast = staticmethod(tf.cast)
+    _transpose = staticmethod(torch.transpose)
+    _tensordot = staticmethod(torch.tensordot)
+    _conj = staticmethod(torch.conj)
+    _imag = staticmethod(torch.imag)
+    _roll = staticmethod(torch.roll)
+    _stack = staticmethod(torch.stack)
+
+    @staticmethod
+    def _asarray(array, dtype=None):
+        try:
+            res = torch.as_tensor(array)
+            if dtype is not None:
+                res = res.type(dtype)
+        except TypeError as e:
+            print("Invalid Argument given to function call \n",e)
+        return res
+
+    def __init__(self, wires, *, shots=None, analytic=None):
+        super().__init__(wires, shots=shots, cache=0, analytic=analytic)
+
+        # # TODO: Need to test this in Torch Env.
+        del self._apply_ops["CZ"]
+
+        # # TODO: Need to test this in Torch Env.
+        if not SUPPORTS_APPLY_OPS or self.num_wires > 8:
+            self._apply_ops = {}
+
+    @classmethod
+    def capabilities(cls):
+        capabilities = super().capabilities().copy()
+        capabilities.update(
+            passthru_interface="pt",
+            supports_reversible_diff=False,
+        )
+        return capabilities
+
+    ## TODO:  this needs to be updated
+    # @staticmethod
+    # def _scatter(indices, array, new_dimensions):
+    #     indices = np.expand_dims(indices, 1)
+    #     return tf.scatter_nd(indices, array, new_dimensions)
```

Comment:
Something like this _might_ work, but I have not tested this or verified it :)

```python
def _scatter(indices, array, new_dimensions):
    new_tensor = torch.zeros(new_dimensions, dtype=array.dtype)
    new_tensor[indices] = array
    return tensor
```

Reply:
kind of stuck on this one, tried following:

```
import torch
indices = torch.tensor([[4], [3], [1], [7]])
updates = torch.tensor([9,10,11,12])
shape2 = (8,0)
new_tensor = torch.zeros((8,0), dtype=updates.dtype)
new_tensor[indices] = updates
print(new_tensor)
```
RuntimeError: shape mismatch: value tensor of shape [4] cannot be broadcast to indexing result of shape [4, 1, 0]